wcfsetup/install/files/lib/data/user/avatar/UserAvatar.class.php

```diff
@@ -55,26 +57,37 @@ class UserAvatar extends DatabaseObject implements IUserAvatar
      * Returns the physical location of this avatar.
      *
      * @param int $size
-     * @return  string
+     * @param bool|null $forceWebP
+     * @return string
      */
-    public function getLocation($size = null)
+    public function getLocation($size = null, ?bool $forceWebP = null)
     {
-        return WCF_DIR . 'images/avatars/' . $this->getFilename($size);
+        return WCF_DIR . 'images/avatars/' . $this->getFilename($size, $forceWebP);
     }
 
     /**
      * Returns the file name of this avatar.
      *
      * @param int $size
-     * @return  string
+     * @param bool|null $forceWebP
+     * @return string
      */
-    public function getFilename($size = null)
+    public function getFilename($size = null, ?bool $forceWebP = null)
     {
-        return \substr(
-            $this->fileHash,
-            0,
-            2
-        ) . '/' . $this->avatarID . '-' . $this->fileHash . ($size !== null ? ('-' . $size) : '') . '.' . $this->avatarExtension;
+        if ($forceWebP === true || ($forceWebP === null && $this->hasWebP && ImageUtil::browserSupportsWebP())) {
```

Comment:
```suggestion
        if (
            $forceWebP === true
            || ($forceWebP === null && $this->hasWebP && ImageUtil::browserSupportsWebP())
       ) {
```
Even if the line length is not exceed this makes it more readable, because each case is handled on a separate line.